src/storage.cpp

```diff
@@ -83,6 +83,88 @@ int Storage::NumberOfImages() {
   return images_->length();
 }
 
+// TODO(all): maybe some refactor to get model by name instead index.
+Reconstruction* Storage::GetReconstruction(int number) {
+  if (status_ != ReconstructionStatus::LOADED_INTO_MEMORY) {
+    ReadReconstructions();
+  }
+
+  return reconstructions_[number];
+}
+
+void Storage::SetReconstructions(
+    const std::vector<Reconstruction*>& reconstructions) {
+  reconstructions_.resize(reconstructions.size());
+  for (int i = 0; i < reconstructions_.size(); i++) {
+    reconstructions_[i] = reconstructions[i];
+  }
+
+  status_ = ReconstructionStatus::LOADED_INTO_MEMORY;
+  LOG(INFO) << "Reconstructions have been saved to memory.";
+}
+
+void Storage::ReadReconstructions() {
+  QDirIterator it(output_location_);
+  std::vector<Reconstruction*> reconstructions;
+
+  LOG(INFO) << "Reading models...";
+  QRegExp rx(MODEL_FILENAME_PATTERN);
+  while (it.hasNext()) {
+    QString next_model;
+    next_model = it.next();
+
+    if (rx.indexIn(next_model) == -1) {
+      LOG(WARNING) << "\t" << next_model.toStdString() <<
+                   "- \"does not match the regex.\"";
+      continue;
+    }
+
+    LOG(INFO) << "\t" << next_model.toStdString();
+    QString filename = QDir(output_location_).filePath(next_model);
+
+    Reconstruction* reconstruction(new Reconstruction());
```

Comment:
how that differs from
`Reconstruction* reconstruction = new Reconstruction()`
?